### ISSUE TITLE
Better VMD integration

### ISSUE THREAD

**Issue by jhenin** [Colvars/colvars#235]
This issue is meant to track the progress of further integration of Colvars, and the Dashboard interface, into VMD. Individual milestones are:

- [x] integrate Dashboard into the standard VMD plugin distribution
- [x] fix for issue #226 
- [x] integrate #220 into VMD codebase
- [x] enable Colvars in official Window builds of VMD
- [x] enable Lepton in Colvars/VMD
- [x] better VMD manual: remove the "Crash course' and introduce Dashboard and command-line / scripts instead.

**Comment by giacomofiorin** [comment]
It may be premature, but I couldn't resist checking the 4th box :-)

**Comment by jhenin** [comment]
A few more boxes checked! One remaining TODO is to test the Mac builds, as soon as a new alpha is out. Then the next step will be Windows...

**Comment by giacomofiorin** [comment]
I just realized that VMD on Mac uses Xcode, not clang.  I'll take a look at the next opportunity.

**Comment by giacomofiorin** [comment]
Updates on this issue: the Windows build (for which I had tentatively checked the 4th box) should be straight forward with a recent enough Visual Studio (I tested VS community 2017 on just the library files).  But it depends on John to include it.

For Lepton, the patch available from the repository below should work OK for Linux and macOS.  Not tested for Windows, as that requires an updated project file for the entire VMD.
https://github.com/giacomofiorin/vmd-patches

**Comment by jhenin** [comment]
21 months later, I checked the 4th box again.

**Comment by giacomofiorin** [comment]
Yeah, that took some time, but it was great to see!  Here is a link to the original email for our records:
https://www.ks.uiuc.edu/Research/vmd/mailing_list/vmd-l/32475.html

Overall, when considering the timeline, seems like VMD 1.9.4 is a moving target despite John's best efforts... See macOS Big Sur, for example.

So I've been wondering what to do for Lepton. Since we first considered proposing John to add it to VMD, Lepton also began requiring C++11. I put a note to that sense in the patches repository:
https://github.com/giacomofiorin/vmd-patches
But I find it unlikely that at this point we can convince John to introduce further significant changes when he also has to deal with manufacturers working against.

Perhaps we need to figure out how to make Lepton itself a dynamically loaded plugin?

**Comment by giacomofiorin** [comment]
Looking back at this issue, the only missing item is actually done:
https://github.com/giacomofiorin/vmd-patches
The patch just isn't in the main VMD distribution, but that is not a Colvars issue at this point.

There are also other enhancements that weren't on this list, like the volmap interface that I added (conversation in #366) and the many very useful hooks that the Dashboard makes with the VMD scripting interface (e1976c0 for example).

It looks to me that all that we could do for this issue, we did.  I see two options with it:
- Keep it open, but then also include all VMD issues that we cannot do much about but aren't resolved yet (e.g. C++11-compatible VMD builds).
- Close it, and have just a separate issue to passively track any changes that John introduces in VMD.
